src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -131,6 +131,7 @@ public DateTimeKind GetCorrespondingKind(TimeZoneInfo? timeZone)
 
             public Dictionary<string, TimeZoneInfo>? _systemTimeZones;
             public ReadOnlyCollection<TimeZoneInfo>? _readOnlySystemTimeZones;
+            public ReadOnlyCollection<TimeZoneInfo>? _readOnlyUnsortedSystemTimeZones;
```

Comment:
Could we instead only keep use one cache to save space? It would cost a boolean flag to check whether or not the collection is sorted by `BaseUtcOffset` -> `DisplayName` instead of another `ReadOnlyCollection<TimeZoneInfo>` with ~428 elements.

The catch would be that the order of the collection would be subject to change only once (initially populating theReadOnlyCollection with a call to `GetSystemTimeZones(true)` then having a call to `GetSystemTimeZones(false)`), bringing up the question, how important is it to maintain an order vs the space savings?

Reply:
Instead of using Boolean I kept `_readOnlyUnsortedSystemTimeZones` and I forced it to be equal to `_readOnlySystemTimeZones` when initializing `_readOnlySystemTimeZones`.  It doesn't matter if the order of `_readOnlyUnsortedSystemTimeZones` change as we always claim the list is not necessarily be sorted so users shouldn't assume any order anyway.